### DISCUSSION TITLE
How can I return a different status code from getStaticProps?

### DISCUSSION THREAD

**Discussion by agamm** [vercel/next.js#45392] [Help]
If I have an error happening in my getStaticProps I want to return a `500` error and not `404` not found. Sometimes I would even want to return something like `204` no content.

How can I do that?

**Comment by icyJoseph** [comment]
`GetStaticProps` runs at build time, and with ISR on, it runs isolated in the server, where it just passes its result back to the server, to be used for page creation.

It has no concept of network statuses really. It is the server that handles statuses. If `GSP` crashes out, nothing happens, but when `GetServerSideProps` crashes, you can see a 500 from the server. During build time if `GSP` crashes out and the build fails.

When you return redirect, or not-found, the server takes care of resolving that. 

You might, redirect users with a 307/308 to an API route that can return the specific error, or status code you want to show.
